### PR TITLE
*: support MockAutoScaler and AWSAutoScaler

### LINKED ISSUE #40747: Support fetch topo information from AutoScaler
## Enhancement
**AutoScaler** will manage all tiflash_compute nodes instead of PD. TiDB needs to fetch topo(ip:port of tiflash_compute nodes) from AutoScaler when dispatch MPPTask.

So need to implement topoFetcher in TiDB. Basically, fetching topo is just a http request.

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #40747

Problem Summary:

**AutoScaler** will manage all tiflash_compute nodes instead of PD. TiDB needs to fetch topo(ip:port of tiflash_compute nodes) from AutoScaler when dispatch MPPTask.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/tiflashcompute/topo_fetcher.go [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+// GetAutoScalerType return topo fetcher type.
+func GetAutoScalerType(typ string) int {
+	switch typ {
+	case MockASStr:
+		return MockASType
+	case AWSASStr:
+		return AWSASType
+	case GCPASStr:
+		return GCPASType
+	case TestASStr:
```
Comment (guo-shaoge) [inline util/tiflashcompute/topo_fetcher.go:104]: todo: remove TestASStr

**Comment by bestwoody** [comment]
LGTM， the real test on dev tier and custom AWS env looks work as expected。

**Review by bestwoody — APPROVED** [review]

**Comment by bestwoody** [comment]
“**The bot only counts approvals from reviewers and higher roles** in [list](https://prow.tidb.net/tichi/repos/pingcap/tidb/pulls/40729/owners), but you're still welcome to leave your comments.”

**Review by wshwsh12 — APPROVED** [review]

File: store/copr/batch_coprocessor.go [hunk 2]
```diff
@@ -515,8 +523,28 @@ func buildBatchCopTasksForPartitionedTable(bo *backoff.Backoffer,
 	return batchTasks, nil
 }
 
-func filterAliveStores(ctx context.Context, stores []*tikv.Store, ttl time.Duration, kvStore *kvStore) []*tikv.Store {
-	var aliveStores []*tikv.Store
+func filterAliveStoresStr(ctx context.Context, storesStr []string, ttl time.Duration, kvStore *kvStore) (aliveStores []string) {
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply store/copr/batch_coprocessor.go:526]: Done

File: util/tiflashcompute/topo_fetcher.go [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+	awsFixedPoolHTTPPath    = "sharedfixedpool"
+	awsFetchHTTPPath        = "resume-and-get-topology"
+	httpGetFailedErrMsg     = "get tiflash_compute topology failed"
+	parseTopoTSFailedErrMsg = "parse timestamp of tiflash_compute topology failed"
+)
+
+var errTopoFetcher = dbterror.ClassUtil.NewStd(errno.ErrInternal)
+
+// TopoFetcher is interface for fetching topo from AutoScaler.
+// There are three kinds of AutoScaler for now:
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply util/tiflashcompute/topo_fetcher.go:77]: Done

File: util/tiflashcompute/topo_fetcher.go [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+	u := url.URL{
+		Scheme:   "http",
+		Host:     f.addr,
+		Path:     awsFetchHTTPPath,
+		RawQuery: para.Encode(),
+	}
+	url := u.String()
+	logutil.BgLogger().Info("fetchTopo", zap.Any("url", url))
+
+	newTopo, err := awsHTTPGetAndParseResp(url)
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply util/tiflashcompute/topo_fetcher.go:427]: fetchTopo() and fetchFixedPoolTopo() have different http parameters.

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**